docs/data/routes/docs/client-frameworks/angular/angular-tips/en.md

```diff
@@ -120,8 +120,40 @@ There aren't changes in your component module. If you use multiple components la
     JssModule.withComponents([
       // non-lazy components
     ], [
-      { path: 'FirstComponent', loadChildren: './my/my.module#MyModule'},
-      { path: 'SecondComponent', loadChildren: './my/my.module#MyModule'},
+      { path: 'FirstComponent', loadChildren: import('./my/my.module').then(mod => mod.MyModule)},
+      { path: 'SecondComponent', loadChildren: import('./my/my.module').then(mod => mod.MyModule)},
+    ]),
+  ],
+  exports: [
+    JssModule,
+    AppComponentsSharedModule,
+  ],
+  declarations: [
+    // non-lazy components
+  ],
+})
+export class AppComponentsModule { }
+```
+
+Components can also have a  for loading using canActivate and canResolve. Classes can be provided as well, they should be derived from JssCanActivate and/or JssResolve.
+
+```js
+@NgModule({
+  imports: [
+    AppComponentsSharedModule,
+    JssModule.withComponents([
+      // non-lazy components
+    ], [
+      { 
+        path: 'FirstComponent', 
+        loadChildren: import('./my/my.module').then(mod => mod.MyModule),
+        canActivate: (input: GuardInput) => { return of(true) }
```

Comment:
@jlast I can't use this example, because types are invalid, can you provide a valid example?
```
Type '(input: GuardInput) => any' is not assignable to type 'JssCanActivate | Type<JssCanActivate> | (JssCanActivate | Type<JssCanActivate>)[]'.
  Type '(input: GuardInput) => any' is missing the following properties from type '(JssCanActivate | Type<JssCanActivate>)[]'
```

Reply:
@jlast I have to use something like:

```
canActivate: {
        canActivate: (input: GuardInput) => { return of(false) }
}
```

But it's not correctly